pyproject.toml

```diff
@@ -31,6 +31,7 @@ requests-mock = ">=1.9.3"
 pytest-mock = "^3.8.2"
 rstcheck = "^6.1.1"
 ruff = "^0.0.270"
+pip-tools = "^7.1.0"
```

Comment:
aren't we missing `pybuild-deps` (given its usage on `make search-build-requirements`)?

Reply:
Good catch! I added this as a dev dependency and amended the last commit.

I probably missed this because the command in that make target (as copied from quipucords) silences errors and always returns a clean exit code. (`2>/dev/null` ... `|| true`) ☹️

Reply:
For fun, I removed the stderr redirection and `true`, and then I ran `make lock-requirements`.

It spewed out a stack trace at one point but ultimately completed with a normal `0` exit code. Here's a piece of that stack trace:

```
...
cat requirements*.txt | grep -vE '(^ )|(#)' | awk '{print $1}' | sed 's/==/ /' | \
	xargs -P6 -n2 poetry run pybuild-deps find-build-deps | \
	sort -u >> requirements-build.in
Traceback (most recent call last):
  File "/Users/brasmith/Library/Caches/pypoetry/virtualenvs/qpc-XVoVAvnv-py3.11/bin/pybuild-deps", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/Users/brasmith/Library/Caches/pypoetry/virtualenvs/qpc-XVoVAvnv-py3.11/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  File "/Users/brasmith/Library/Caches/pypoetry/virtualenvs/qpc-XVoVAvnv-py3.11/lib/python3.11/site-packages/pybuild_deps/get_package_source.py", line 24, in get_package_source
    return retrieve_and_save_source_from_pypi(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/brasmith/Library/Caches/pypoetry/virtualenvs/qpc-XVoVAvnv-py3.11/lib/python3.11/site-packages/pybuild_deps/get_package_source.py", line 37, in retrieve_and_save_source_from_pypi
    source_url = get_source_url(package_name, version)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/brasmith/Library/Caches/pypoetry/virtualenvs/qpc-XVoVAvnv-py3.11/lib/python3.11/site-packages/pybuild_deps/get_package_source.py", line 52, in get_source_url
    raise NotImplementedError()
NotImplementedError
...
```
😕 

Should we _not_ ignore this?

Reply:
Mmmmm. I'll need to dig the blame on quipucords to see if I had any extra context for a proper answer.

Reply:
lol. no need to check blame... https://github.com/bruno-fs/pybuild-deps/blob/main/src/pybuild_deps/get_package_source.py#L37-L40

Reply:
it's an issue on `pybuild_deps` side... designed to fail like this.

It would be interesting to know which exact dependencies are causing this failure

Reply:
`setuptools-scm[toml]` and `celery[kombu]` are causing this (on quipucords side)

Reply:
replacing the `sed` command with `perl -pe 's/==/ /;s/\[\w+\]//'` suppress the errors on my system

Reply:
sed/grep/awk ftw ❤️